lib/sandthorn/aggregate_root_base.rb

```diff
@@ -150,6 +154,16 @@ def aggregate_build events
           aggregate
         end
 
+        def events(*event_names)
+          event_names.each do |name|
+            define_method(name) do |*args, &block|
+              block.call() if block
```

Comment:
Tried to call it like that but could not get it to work, more info here -> http://stackoverflow.com/questions/2306731/using-yield-inside-define-method-in-ruby